### PR TITLE
10.0.0: remove deprecated features

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Remove deprecated features that are scheduled for removal in 10.0.0.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
consul
ejabberd_user
irc
proxmox_kvm
redhat_subscriptions
rhsm_repository

### CODE REVIEW

**Comment by jpmens** [comment]
shipit

(for `irc`)

**Comment by ptoscano** [comment]
LGTM for `redhat_subscription` and `rhsm_repository`.

(and thanks for also dropping the unused functions)

File: plugins/modules/redhat_subscription.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
     - Since community.general 6.5.0, credentials (that is, O(username) and O(password),
       O(activationkey), or O(token)) are needed only in case the the system is not registered,
       or O(force_register) is specified; this makes it possible to use the module to tweak an
-      already registered system, for example attaching pools to it (using O(pool), or O(pool_ids)),
+      already registered system, for example attaching pools to it (O(pool_ids)),
```
Comment (ptoscano) [inline plugins/modules/redhat_subscription.py:42]: oh here "using" was accidentally dropped

**Comment by felixfontein** [comment]
@russoz there's another deprecation in plugins/module_utils/cmd_runner.py, it might be better if you look at that one in a separate PR.

**Comment by russoz** [comment]
looks like the module `consul_acl` is to be deprecated as well

**Comment by felixfontein** [comment]
> looks like the module `consul_acl` is to be deprecated as well

You mean removed. That happens in #8921, which removes full modules/plugins. This PR only removes parts of modules/plugins.

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
I'll merge this at the weekend if nobody objects. If you want to take a closer look but need more time, please mention it here, then I'll wait with merging :)

**Comment by apollo13** [comment]
LGTM

**Comment by felixfontein** [comment]
@jpmens @ptoscano @russoz @apollo13 thanks a lot for reviewing this!

**Pull request merged**